mods/persistence/modules/modular_computers/file_system/programs/trade_beacon_program.dm

```diff
@@ -0,0 +1,560 @@
+	program_menu_icon = "cart"
+
+	available_on_network = 1
+
+	size = 12
+	nanomodule_path = /datum/nano_module/program/trade_management
+	category = PROG_SUPPLY
+
+	var/obj/effect/overmap/trade_beacon/selected_beacon
+
+	var/import_menu = 1
+	var/export_menu = 0
+	var/log_menu = 0
+	var/datum/beacon_import/selected_import
+	var/datum/beacon_export/selected_export
+	var/payment_type = 1 // 1 == personal account 2 == chargestick 3 == network parent account
+
+	var/list/detected_telepads = list()
+	var/obj/selected_telepad
+
+/datum/computer_file/program/trade_management/proc/detect_telepads()
+	detected_telepads = list()
+	for(var/obj/machinery/telepad_cargo/c in orange(3, get_turf(computer)))
+		if(c.anchored)
+			detected_telepads |= c
+	return
+
+/datum/computer_file/program/trade_management/on_startup(mob/living/user, datum/extension/interactive/os/new_host)
+	. = ..()
+	detect_telepads()
+
+
+/datum/nano_module/program/trade_management
+	name = "Import/Export Management"
+
+
+/datum/computer_file/program/trade_management/proc/spawnImport()
+	var/turf/T = get_turf(selected_telepad)
+	flick("pad-beam", selected_telepad)
+	. = selected_import.spawnImport(T)
+	if(selected_import.remaining_stock < 1)
+		selected_beacon.active_imports.Remove(selected_import)
+		selected_import = null
+/datum/computer_file/program/trade_management/proc/checkImport()
+	var/turf/T = get_turf(selected_telepad)
+	flick("pad-beam", selected_telepad)
+	return selected_import.checkImport(T)
+
+/datum/computer_file/program/trade_management/proc/takeExport()
+	flick("pad-beam", selected_telepad)
+	. = selected_export.takeExport()
+	if(selected_export.remaining_stock < 1)
+		selected_beacon.active_exports.Remove(selected_export)
+		selected_export = null
+/datum/computer_file/program/trade_management/proc/checkExport()
+	var/turf/T = get_turf(selected_telepad)
+	return selected_export.checkExport(T)
+
+/datum/computer_file/program/trade_management/proc/resetExport()
+	return selected_export.resetExport()
+
+
+/datum/nano_module/program/trade_management/ui_interact(mob/user, ui_key, datum/nanoui/ui, force_open, datum/nanoui/master_ui, datum/topic_state/state)
+	var/list/data = host.initial_data()
+	var/datum/computer_file/program/trade_management/PRG = program
+	var/datum/extension/interactive/os/computer = PRG.computer
+	var/list/telepads[0]
+	var/update_time = time2text((SStrade_beacons.last_cycle+2 HOURS),"hh:mm")
+	for(var/obj/machinery/telepad_cargo/x in PRG.detected_telepads)
+		var/selected = 0
+		if(x == PRG.selected_telepad) selected = 1
+		telepads.Add(list(list(
+			"telepad_name" = "[x.name] ([x.telepad_id])",
+			"selected" = selected,
+			"telepad_ref" = any2ref(x),
+		)))
+	if(!telepads.len)
+		telepads.Add(list(list(
+			"telepad_name" = "No telepads detected!",
+			"selected" = 1,
+		)))
+	data["telepads"] = telepads
+	data["update_time"] = update_time
+	var/obj/effect/overmap/trade_beacon/selected_beacon = PRG.selected_beacon
+	if(selected_beacon)
```

Comment:
It would be nice if you could split that big if block into a few procs so it's a bit easier to read. Nothing blocking, just a suggestion.

Reply:
I would agree with you normally, but in the context of ui_interact and the function of data[], I want to keep it as is. For context, the ui_interact is layed out in order of the template file and they are meant to be viewed together